hardhat/app/src/main.js

```diff
@@ -4,13 +4,12 @@ import { createApp } from 'vue'
 import App from '@/App'
 import { installEthers } from '@/plugin'
 
-const main = async () => {
+const main = () => {
   const provider = new ethers.providers.Web3Provider(window.ethereum)
-  await provider.send('eth_requestAccounts', [])
 
-  createApp(App)
+  createApp(App, { provider: provider })
```

Comment:
Why the second argument? By the [docs][1], it is passed as `props` to the root component. But I did not see it being used

[1]: https://vuejs.org/api/application.html#createapp

Reply:
Ah yea, passed that in and used it but then after refactored it out and connected to metamask directly with the MetaMask API directly through window.ethereum.
Also wasn't sure about whether the ethers plugin should be accessible through this.$ethers or something .

Reply:
To make it accessible via `$ethers` you must install it via the [plugin][1] API (the only way I know how)

Quick and dirty:

```js
const provider = new Provider()

const plugin = {
  install (app, ethers) {
    app.config.globalProperties.$ethers = ethers
  }
}

app.use(plugin, provider)
```

[1]: https://vuejs.org/guide/reusability/plugins.html